pkg/kubelet/watchdog/watchdog_linux.go

```diff
@@ -104,25 +99,39 @@ func NewHealthChecker(syncLoop syncLoopHealthChecker, opts ...Option) (HealthChe
 		return nil, fmt.Errorf("configure watchdog timeout too small: %v", watchdogVal)
 	}
 
-	// The health checks performed by checkers are the same as those for "/healthz".
-	checkers := []healthz.HealthChecker{
-		healthz.PingHealthz,
-		healthz.LogHealthz,
-		healthz.NamedCheck("syncloop", syncLoop.SyncLoopHealthCheck),
-	}
 	retryBackoff := wait.Backoff{
 		Duration: time.Second,
 		Factor:   2.0,
 		Jitter:   0.1,
 		Steps:    2,
 	}
-	hc.checkers = append(hc.checkers, checkers...)
 	hc.retryBackoff = retryBackoff
 	hc.interval = watchdogVal / 2
 
 	return hc, nil
 }
 
+func (hc *healthChecker) SetHealthCheckers(syncLoop syncLoopHealthChecker, checkers []healthz.HealthChecker) {
+	// Define the default set of health checkers that should always be present
+	defaultCheckers := []healthz.HealthChecker{
+		healthz.PingHealthz,
+		healthz.LogHealthz,
+		healthz.NamedCheck("syncloop", syncLoop.SyncLoopHealthCheck),
+	}
+
+	var combined []healthz.HealthChecker
+	combined = append(combined, defaultCheckers...)
+	combined = append(combined, checkers...)
+	hc.checkers.Store(combined)
```

Comment:
if we assign the whole collection here, does it need to be atomic? Since we are not modifying the collection that another thread may be iterating, than just assignment may be fine?